File: src/main/java/jp/artan/japanesefoodmod/raw_materials/wood/RawMaterialsFruitTree.java
package jp.artan.japanesefoodmod.raw_materials.wood;

import jp.artan.japanesefoodmod.common.wood.FruitTree;
import jp.artan.japanesefoodmod.common.world.WorldGenSupplier;
import jp.artan.japanesefoodmod.raw_materials.food.RawMaterialFood;

public class RawMaterialsFruitTree extends FruitTree {

    public RawMaterialsFruitTree(String name, int amount, float saturation, WorldGenSupplier supplier) {
        super(new RawMaterialFood(name, amount, saturation), new RawMaterialsFruitLog(name),
                new RawMaterialsFruitLeaves(name), new RawMaterialsFruitSapling(name, supplier),
                new RawMaterialsFruitBlock(name));
        this.leaves.setItemDropped(this.sapling);
        this.fruitBlock.setFruitItem(this.fruit);
        supplier.setIBlockState(this.log, this.leaves, this.fruitBlock);
    }
}

File: src/main/java/jp/artan/japanesefoodmod/raw_materials/wood/RawMaterialsFruitTree.kt
package jp.artan.japanesefoodmod.raw_materials.wood

import jp.artan.japanesefoodmod.common.wood.FruitTree
import jp.artan.japanesefoodmod.common.world.tree.WorldGenSupplier
import jp.artan.japanesefoodmod.raw_materials.food.RawMaterialFood

class RawMaterialsFruitTree(name: String, amount: Int, saturation: Float, supplier: WorldGenSupplier) :
        FruitTree(RawMaterialFood(name, amount, saturation), RawMaterialsFruitLog(name), RawMaterialsFruitLeaves(name),
                RawMaterialsFruitSapling(name, supplier), RawMaterialsFruitBlock(name)) {
    init {
        this.leaves.sapling = this.sapling
        this.fruitBlock.fruitItem = this.fruit
        supplier.setIBlockState(this.log, this.leaves, this.fruitBlock)
    }
}
